### PR TITLE
Validate that all `go.mod` files are registered in `DEFAULT_MODULES`

### PR DESCRIPTION
**Pull request by FlorentClarret**
### What does this PR do?

Validate that all `go.mod` files are registered in `DEFAULT_MODULES`. We do not need the previous logic anymore since we check every single module in the repo.

### Motivation

We noticed some modules are not registered, so they are not validated by the `tidy` command in the CI. 

Because of that @wdhif had to fix it [afterward](https://github.com/DataDog/datadog-agent/pull/26771) 

### Additional Notes

- https://datadoghq.atlassian.net/browse/ADXT-358
- The Otel team will remove the otel module next week
- This new implementation is super fast which allows us to add it as a pre-commit hook without any issues

### CODE REVIEW

File: tasks/modules.py [hunk 1]
```diff
@@ -366,19 +382,25 @@ def for_each(ctx: Context, cmd: str, skip_untagged: bool = False):
...
-            if dependency not in DEFAULT_MODULES:
-                if fail_fast:
-                    raise Exit(f"{color_message('ERROR', 'red')}: {module.path} depends on missing {dependency}")
-                missing_modules.append((module, dependency))
+    missing_modules: list[str] = []
+
+    # Find all go.mod files and make sure they are registered in DEFAULT_MODULES
+    for root, dirs, files in os.walk("."):
+        # Ignore the files that could be stored in the Python tests
+        dirs[:] = [d for d in dirs if root + '/' + d != "./tasks"]
```
Comment (CelianR) [inline tasks/modules.py:394]: Is it equivalent to this ?
```suggestion
        dirs.remove(root + '/' + d)
```

Comment (FlorentClarret) [reply tasks/modules.py:394]: `d` is part of the list, you can't use it that way

Comment (FlorentClarret) [reply tasks/modules.py:394]: but `dirs.remove("./tasks")` would?

Comment (FlorentClarret) [reply tasks/modules.py:394]: no it won't because I need `root`. And I can't just remove `tasks` because I want to check the full path

Comment (CelianR) [reply tasks/modules.py:394]: Indeed it should be the same as `if root == '.': dirs.remove('tasks')` isn't ?

**Review by pducolin — APPROVED** [review]

File: tasks/modules.py (hunk 1 above)
Comment (FlorentClarret) [reply tasks/modules.py]: what do you think of https://github.com/DataDog/datadog-agent/pull/26931/commits/778da6148c98a8d2951b45f630f57f0886ea2d91?

**Review by CelianR — APPROVED** [review]

**Pull request merged**